src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadDocument.java

```diff
@@ -10,5 +12,21 @@
  * @param id Document Id identifying the document
  * @param txnId Unique UUID resenting point in time of a document, used for LWT transactions
  * @param document JsonNode representation of the document
+ * @param sortColumns List<JsonNode> Serialized sort column value
+ * @param docJsonValue Grpc column value for doc_json.
  */
-public record ReadDocument(DocumentId id, UUID txnId, JsonNode document) {}
+public record ReadDocument(
+    DocumentId id,
+    UUID txnId,
+    JsonNode document,
+    List<JsonNode> sortColumns,
+    QueryOuterClass.Value docJsonValue) {
```

Comment:
Is there a reason for not passing `String` but raw gRPC value here? (perhaps we don't always need it so less conversions?)

Reply:
Idea is to deserialize the doc_json only for the selected documents that are returned in response.

Reply:
Ok that makes sense then, thanks!

Reply:
Then we should have used the `Supplier<JsonNode>` or what ever the generic typing would be.. I also very much dislike that we have the grpc here..

Reply:
I think simply extracting `String` would not add much overhead; it's parsing `String`/`byte[]` to `JsonNode` that is heavy. Could do supplier too if that's cleaner of course. Either way, avoiding constructing `JsonNode` makes sense, and so does not passing gRPC type.